diff --git a/quartz/cfg.ts b/quartz/cfg.ts
index 57dff5c759b23b1d11359610553465280160eb2f..5743aa1672c2e913653f53f8c479f973709f8dd5 100644
--- a/quartz/cfg.ts
+++ b/quartz/cfg.ts
@@ -18,6 +18,7 @@ export type Analytics =
       provider: "umami"
       websiteId: string
       host?: string
+      scriptName?: string
     }
   | {
       provider: "goatcounter"
diff --git a/quartz/components/Comments.tsx b/quartz/components/Comments.tsx
index a7315214f35cef55292aa15f26a61506ce69da8b..a5dc7b5881487443c7487e7005c1fdc67b91a0a1 100644
--- a/quartz/components/Comments.tsx
+++ b/quartz/components/Comments.tsx
@@ -46,11 +46,7 @@ export default ((opts: Options) => {
         data-strict={boolToStringBool(opts.options.strict ?? true)}
         data-reactions-enabled={boolToStringBool(opts.options.reactionsEnabled ?? true)}
         data-input-position={opts.options.inputPosition ?? "bottom"}
-        data-light-theme={opts.options.lightTheme ?? "light"}
-        data-dark-theme={opts.options.darkTheme ?? "dark"}
-        data-theme-url={
-          opts.options.themeUrl ?? `https://${cfg.baseUrl ?? "example.com"}/static/giscus`
-        }
+        data-theme="preferred_color_scheme"
         data-lang={opts.options.lang ?? "en"}
       ></div>
     )
diff --git a/quartz/components/ContentMeta.tsx b/quartz/components/ContentMeta.tsx
index e378bcceed3958417ddb1519cbe42a362b5d2274..efd6be1fc4f6c104a8f6f4b49f81b3a2684b0253 100644
--- a/quartz/components/ContentMeta.tsx
+++ b/quartz/components/ContentMeta.tsx
@@ -12,6 +12,7 @@ interface ContentMetaOptions {
    */
   showReadingTime: boolean
   showComma: boolean
+  repoBlobLink?: string
 }
 
 const defaultOptions: ContentMetaOptions = {
@@ -30,7 +31,16 @@ export default ((opts?: Partial<ContentMetaOptions>) => {
       const segments: (string | JSX.Element)[] = []
 
       if (fileData.dates) {
-        segments.push(<Date date={getDate(cfg, fileData)!} locale={cfg.locale} />)
+        const createdDate = fileData.dates.created
+        const modifiedDate = fileData.dates.modified
+        segments.push(<>
+          {i18n(cfg.locale).components.contentMeta.created}: <Date date={createdDate} locale={cfg.locale} />
+        </>)
+        if (modifiedDate && modifiedDate.getTime() !== createdDate.getTime()) {
+          segments.push(<>
+            {i18n(cfg.locale).components.contentMeta.modified}: <Date date={modifiedDate} locale={cfg.locale} />
+          </>)
+        }
       }
 
       // Display reading time if enabled
@@ -42,6 +52,11 @@ export default ((opts?: Partial<ContentMetaOptions>) => {
         segments.push(<span>{displayedTime}</span>)
       }
 
+      if (options.repoBlobLink) {
+        const pathInsideContent = fileData.filePath!.replace(/^content\//, "")
+        segments.push(<a href={`${options.repoBlobLink}/${pathInsideContent}?plain=1`}>{i18n(cfg.locale).components.contentMeta.source}</a>)
+      }
+
       return (
         <p show-comma={options.showComma} class={classNames(displayClass, "content-meta")}>
           {segments}
diff --git a/quartz/components/Graph.tsx b/quartz/components/Graph.tsx
index 907372e9375387392b45b12a7af4255a84570500..37edd51d465dced4c68898d6c0967abfce20ca5a 100644
--- a/quartz/components/Graph.tsx
+++ b/quartz/components/Graph.tsx
@@ -68,7 +68,8 @@ export default ((opts?: Partial<GraphOptions>) => {
         <h3>{i18n(cfg.locale).components.graph.title}</h3>
         <div class="graph-outer">
           <div class="graph-container" data-cfg={JSON.stringify(localGraph)}></div>
-          <button class="global-graph-icon" aria-label="Global Graph">
+          {/* comply to buggy getElementById inside floatingButtons.inline.ts */}
+          <button class="global-graph-icon" id="global-graph-icon" aria-label="Global Graph">
             <svg
               version="1.1"
               xmlns="http://www.w3.org/2000/svg"
diff --git a/quartz/components/Head.tsx b/quartz/components/Head.tsx
index 23183ca8c23810407e2506dcd418d0f71184aa07..79f9a3f083cb9f9233e80bd3ad72eddca9a3dbc7 100644
--- a/quartz/components/Head.tsx
+++ b/quartz/components/Head.tsx
@@ -50,7 +50,6 @@ export default (() => {
             )}
           </>
         )}
-        <link rel="preconnect" href="https://cdnjs.cloudflare.com" crossOrigin="anonymous" />
         <meta name="viewport" content="width=device-width, initial-scale=1.0" />
 
         <meta name="og:site_name" content={cfg.pageTitle}></meta>
diff --git a/quartz/components/RecentNotes.tsx b/quartz/components/RecentNotes.tsx
index 2c32feadf2df77ea2ac6cfa0ddc7fd207ed0ca47..9d7ee700a5c7ecd8c6c387a1f875384e26b6de68 100644
--- a/quartz/components/RecentNotes.tsx
+++ b/quartz/components/RecentNotes.tsx
@@ -12,6 +12,7 @@ interface Options {
   title?: string
   limit: number
   linkToMore: SimpleSlug | false
+  linkToHistory?: string
   showTags: boolean
   filter: (f: QuartzPluginData) => boolean
   sort: (f1: QuartzPluginData, f2: QuartzPluginData) => number
@@ -37,7 +38,13 @@ export default ((userOpts?: Partial<Options>) => {
     const remaining = Math.max(0, pages.length - opts.limit)
     return (
       <div class={classNames(displayClass, "recent-notes")}>
-        <h3>{opts.title ?? i18n(cfg.locale).components.recentNotes.title}</h3>
+        <h3>{opts.title ?? i18n(cfg.locale).components.recentNotes.title}
+          {opts.linkToMore && (
+            <span class="see-more">
+              <a href={opts.linkToHistory} class="external see-more">{i18n(cfg.locale).components.recentNotes.seeHistory}</a>
+            </span>
+          )}
+        </h3>
         <ul class="recent-ul">
           {pages.slice(0, opts.limit).map((page) => {
             const title = page.frontmatter?.title ?? i18n(cfg.locale).propertyDefaults.title
diff --git a/quartz/components/styles/listPage.scss b/quartz/components/styles/listPage.scss
index e86c39dcb90abd3bdf6dffcecc9ddb4cb57dfc53..79fd71f4fcddb68b4392dba5fd759e4df35caffa 100644
--- a/quartz/components/styles/listPage.scss
+++ b/quartz/components/styles/listPage.scss
@@ -11,7 +11,8 @@ li.section-li {
 
   & > .section {
     display: grid;
-    grid-template-columns: fit-content(8em) 3fr 1fr;
+    // 使用固定值同步列表每个元素的宽度
+    grid-template-columns: 9em 3fr 1fr;
 
     @media all and ($mobile) {
       & > .tags {
diff --git a/quartz/components/styles/recentNotes.scss b/quartz/components/styles/recentNotes.scss
index 726767198be13cbbd02c3ac912a0f4a4dd89ad0b..e254a100befe5c0107d201d6db1f236c1fc53a09 100644
--- a/quartz/components/styles/recentNotes.scss
+++ b/quartz/components/styles/recentNotes.scss
@@ -21,4 +21,11 @@
       }
     }
   }
+
+}
+.see-more {
+  font-size: 0.85rem;
+  opacity: 0.5;
+  white-space: pre;
+  display: inline-block;
 }
diff --git a/quartz/i18n/locales/definition.ts b/quartz/i18n/locales/definition.ts
index f22538fca336b24fda0ac1d828568ee5a87e3022..7283dae5893c248304a515b97763d07f58cf8edb 100644
--- a/quartz/i18n/locales/definition.ts
+++ b/quartz/i18n/locales/definition.ts
@@ -47,6 +47,7 @@ export interface Translation {
     recentNotes: {
       title: string
       seeRemainingMore: (variables: { remaining: number }) => string
+      seeHistory: string
     }
     transcludes: {
       transcludeOf: (variables: { targetSlug: FullSlug }) => string
@@ -61,6 +62,9 @@ export interface Translation {
     }
     contentMeta: {
       readingTime: (variables: { minutes: number }) => string
+      created: string
+      modified: string
+      source: string
     }
   }
   pages: {
diff --git a/quartz/i18n/locales/en-US.ts b/quartz/i18n/locales/en-US.ts
index e1111e9ab1791d2ac548e279dd9ea2ead2485325..9258be71b326ec987a67bab266de752b9c27b9ab 100644
--- a/quartz/i18n/locales/en-US.ts
+++ b/quartz/i18n/locales/en-US.ts
@@ -44,6 +44,7 @@ export default {
     recentNotes: {
       title: "Recent Notes",
       seeRemainingMore: ({ remaining }) => `See ${remaining} more →`,
+      seeHistory: "See History →",
     },
     transcludes: {
       transcludeOf: ({ targetSlug }) => `Transclude of ${targetSlug}`,
@@ -58,6 +59,9 @@ export default {
     },
     contentMeta: {
       readingTime: ({ minutes }) => `${minutes} min read`,
+      created: "Created",
+      modified: "Modified",
+      source: "Source Code",
     },
   },
   pages: {
diff --git a/quartz/i18n/locales/zh-CN.ts b/quartz/i18n/locales/zh-CN.ts
index 09951bbb31ff9bb2666854f25a4107fa6d395632..6e2e90ff26c40359636e3ff41a0e98614d1f51a5 100644
--- a/quartz/i18n/locales/zh-CN.ts
+++ b/quartz/i18n/locales/zh-CN.ts
@@ -44,6 +44,7 @@ export default {
     recentNotes: {
       title: "最近的笔记",
       seeRemainingMore: ({ remaining }) => `查看更多${remaining}篇笔记 →`,
+      seeHistory: "查看历史记录 →",
     },
     transcludes: {
       transcludeOf: ({ targetSlug }) => `包含${targetSlug}`,
@@ -58,6 +59,9 @@ export default {
     },
     contentMeta: {
       readingTime: ({ minutes }) => `${minutes}分钟阅读`,
+      created: "创建于",
+      modified: "更新于",
+      source: "本页源码",
     },
   },
   pages: {
diff --git a/quartz/plugins/emitters/componentResources.ts b/quartz/plugins/emitters/componentResources.ts
index f97fddec28e5265e74b249e76299f90095eb4511..4535f9355d853c40ef8d938f5bbc9a88361e9dc7 100644
--- a/quartz/plugins/emitters/componentResources.ts
+++ b/quartz/plugins/emitters/componentResources.ts
@@ -126,7 +126,7 @@ function addGlobalPageResources(ctx: BuildCtx, componentResources: ComponentReso
   } else if (cfg.analytics?.provider === "umami") {
     componentResources.afterDOMLoaded.push(`
       const umamiScript = document.createElement("script");
-      umamiScript.src = "${cfg.analytics.host ?? "https://analytics.umami.is"}/script.js";
+      umamiScript.src = "${cfg.analytics.host ?? "https://analytics.umami.is"}/${cfg.analytics.scriptName ?? "script.js"}";
       umamiScript.setAttribute("data-website-id", "${cfg.analytics.websiteId}");
       umamiScript.setAttribute("data-auto-track", "true");
       umamiScript.defer = true;
diff --git a/quartz/plugins/emitters/ogImage.tsx b/quartz/plugins/emitters/ogImage.tsx
index 813d9348c2ad8c131d8d8f707e70d2f25c066ace..63893c19084fab01a80218ac5d16c18b86010feb 100644
--- a/quartz/plugins/emitters/ogImage.tsx
+++ b/quartz/plugins/emitters/ogImage.tsx
@@ -62,7 +62,7 @@ async function generateSocialImage(
     },
   })
 
-  return sharp(Buffer.from(svg)).webp({ quality: 40 })
+  return sharp(Buffer.from(svg)).webp({ quality: 60 })
 }
 
 async function processOgImage(
diff --git a/quartz/plugins/transformers/frontmatter.ts b/quartz/plugins/transformers/frontmatter.ts
index 1103900c56cd7f7dc70ba86af9ffb436d0903d13..ff79301d3e994c01c3c3848e849520a60fa0943d 100644
--- a/quartz/plugins/transformers/frontmatter.ts
+++ b/quartz/plugins/transformers/frontmatter.ts
@@ -10,11 +10,17 @@ import { i18n } from "../../i18n"
 export interface Options {
   delimiters: string | [string, string]
   language: "yaml" | "toml"
+  created_properties: string[]
+  modified_properties: string[]
+  published_properties: string[]
 }
 
 const defaultOptions: Options = {
   delimiters: "---",
   language: "yaml",
+  created_properties: ["created", "date"],
+  modified_properties: ["modified", "lastmod", "updated", "last-modified"],
+  published_properties: ["published", "publishDate", "date"],
 }
 
 function coalesceAliases(data: { [key: string]: any }, aliases: string[]) {
@@ -100,20 +106,15 @@ export const FrontMatter: QuartzTransformerPlugin<Partial<Options>> = (userOpts)
 
             const socialImage = coalesceAliases(data, ["socialImage", "image", "cover"])
 
-            const created = coalesceAliases(data, ["created", "date"])
+            const created = coalesceAliases(data, opts.created_properties)
             if (created) {
               data.created = created
               data.modified ||= created // if modified is not set, use created
             }
-
-            const modified = coalesceAliases(data, [
-              "modified",
-              "lastmod",
-              "updated",
-              "last-modified",
-            ])
+            
+            const modified = coalesceAliases(data, opts.modified_properties)
             if (modified) data.modified = modified
-            const published = coalesceAliases(data, ["published", "publishDate", "date"])
+            const published = coalesceAliases(data, opts.published_properties)
             if (published) data.published = published
 
             if (socialImage) data.socialImage = socialImage
diff --git a/quartz/plugins/transformers/htmlhead.tsx b/quartz/plugins/transformers/htmlhead.tsx
new file mode 100644
index 0000000000000000000000000000000000000000..2288a38c0167de4bfa93a6afd6909673e69908d8
--- /dev/null
+++ b/quartz/plugins/transformers/htmlhead.tsx
@@ -0,0 +1,47 @@
+import { JSX } from "preact/jsx-runtime"
+import { QuartzTransformerPlugin } from "../types"
+
+/** 往 JSX 中添加 onload 属性 */
+function preloadStylesheetOnloadFn() {
+  return { 'onload': "this.onload=null;this.media='all'" };
+}
+
+function misansFontStylesheet(): JSX.Element {
+  return (
+    <>
+      <link rel="stylesheet" media="print" crossorigin="anonymous"
+        {...preloadStylesheetOnloadFn()}
+        href="https://registry.npmmirror.com/misans/4.1.0/files/lib/Normal/MiSans-Medium.min.css" />
+      <link rel="stylesheet" media="print" crossorigin="anonymous"
+        {...preloadStylesheetOnloadFn()}
+        href="https://registry.npmmirror.com/misans/4.1.0/files/lib/Normal/MiSans-Bold.min.css" />
+    </>
+  )
+}
+
+function jetbrainsMapleMonoFontStylesheet(): JSX.Element {
+  return (
+    <>
+      <link rel="stylesheet" media="print" crossorigin="anonymous"
+        {...preloadStylesheetOnloadFn()}
+        href="https://fontsapi.zeoseven.com/521/main/result.css" />
+      <link rel="stylesheet" media="print" crossorigin="anonymous"
+        {...preloadStylesheetOnloadFn()}
+        href="https://fontsapi.zeoseven.com/521/bold/result.css" />
+    </>
+  )
+}
+
+export const HtmlHead: QuartzTransformerPlugin = () => {
+  return {
+    name: "HtmlHead",
+    externalResources() {
+      return {
+        additionalHead: [
+          misansFontStylesheet(),
+          jetbrainsMapleMonoFontStylesheet(),
+        ]
+      }
+    },
+  }
+}
diff --git a/quartz/util/og.tsx b/quartz/util/og.tsx
index 2afd606273e143e9ea8759bd61c4e59b3d09d420..e1527aec6aaa18edfe413ebf5deb0e08af4ecfd3 100644
--- a/quartz/util/og.tsx
+++ b/quartz/util/og.tsx
@@ -88,6 +88,15 @@ export async function fetchTtf(
     // ignore errors and fetch font
   }
 
+  let fontUrl: string;
+  if (rawFontName === "MiSans") {
+    const weightNameMap: Record<number, string> = {
+      400: "Regular",
+      700: "Bold",
+    }
+    const filename = `MiSans-${weightNameMap[weight]}.ttf`
+    fontUrl = `https://wget.la/https://github.com/dsrkafuu/misans/blob/main/raw/Normal/ttf/${filename}`
+  } else {
   // Get css file from google fonts
   const cssResponse = await fetch(
     `https://fonts.googleapis.com/css2?family=${fontName}:wght@${weight}`,
@@ -107,9 +116,11 @@ export async function fetchTtf(
     )
     return
   }
+  fontUrl = match[1];
+  }
 
   // fontData is an ArrayBuffer containing the .ttf file data
-  const fontResponse = await fetch(match[1])
+  const fontResponse = await fetch(fontUrl)
   const fontData = Buffer.from(await fontResponse.arrayBuffer())
   await fs.mkdir(cacheDir, { recursive: true })
   await fs.writeFile(cachePath, fontData)
diff --git a/quartz/util/theme.ts b/quartz/util/theme.ts
index ff4453b9f33d835c18a9b443ecb0ea9c336b1d57..f5282e6ffd37f735b0cef56b03a963762ec2159f 100644
--- a/quartz/util/theme.ts
+++ b/quartz/util/theme.ts
@@ -1,3 +1,6 @@
+import { get } from "http"
+import { escape } from "querystring"
+
 export interface ColorScheme {
   light: string
   lightgray: string
@@ -21,6 +24,7 @@ export type FontSpecification =
       name: string
       weights?: number[]
       includeItalic?: boolean
+      extras?: string[]
     }
 
 export interface Theme {
@@ -49,6 +53,14 @@ export function getFontSpecificationName(spec: FontSpecification): string {
   return spec.name
 }
 
+function getFontSpecificationExtraNames(spec: FontSpecification): string[] {
+  if (typeof spec === "string") {
+    return []
+  }
+
+  return spec.extras ?? []
+}
+
 function formatFontSpecification(
   type: "title" | "header" | "body" | "code",
   spec: FontSpecification,
@@ -141,6 +153,11 @@ export async function processGoogleFonts(
 }
 
 export function joinStyles(theme: Theme, ...stylesheet: string[]) {
+  const codeFonts = [
+    getFontSpecificationName(theme.typography.code),
+    ...getFontSpecificationExtraNames(theme.typography.code),
+  ].map(name => JSON.stringify(name)).join(", ")
+
   return `
 ${stylesheet.join("\n\n")}
 
@@ -158,7 +175,7 @@ ${stylesheet.join("\n\n")}
   --titleFont: "${getFontSpecificationName(theme.typography.title || theme.typography.header)}", ${DEFAULT_SANS_SERIF};
   --headerFont: "${getFontSpecificationName(theme.typography.header)}", ${DEFAULT_SANS_SERIF};
   --bodyFont: "${getFontSpecificationName(theme.typography.body)}", ${DEFAULT_SANS_SERIF};
-  --codeFont: "${getFontSpecificationName(theme.typography.code)}", ${DEFAULT_MONO};
+  --codeFont: ${codeFonts}, ${DEFAULT_MONO};
 }
 
 :root[saved-theme="dark"] {
